CFC_DataCollector/modeinfer/pipeline.py

```diff
@@ -83,20 +87,43 @@ def getSectionQueryWithGroundTruth(groundTruthMode):
 
   # TODO: Refactor into generic steps and results
   def loadTrainingDataStep(self, sectionQuery, sectionDb = None):
+    logging.debug("START TRAINING DATA STEP %s" % (datetime.now()))
     if (sectionDb == None):
       sectionDb = self.Sections
+
+    begin = time.time()
     logging.debug("Section data set size = %s" % sectionDb.find({'type': 'move'}).count())
+    duration = time.time() - begin
+    logging.debug("Getting dataset size took %s" % (duration))
+        
+    logging.debug("Querying confirmedSections %s" % (datetime.now()))
+    begin = time.time()
     confirmedSections = sectionDb.find(sectionQuery)
+    duration = time.time() - begin
+    logging.debug("Querying confirmedSection took %s" % (duration))
+    
+    logging.debug("Querying stage modes %s" % (datetime.now()))
+    begin = time.time()
     modeList = []
     for mode in MongoClient('localhost').Stage_database.Stage_Modes.find():
         modeList.append(mode)
         logging.debug(mode)
+    duration = time.time() - begin
+    logging.debug("Querying stage modes took %s" % (duration))
```

Comment:
It would be nice to put this into a separate function.

def logtime(function):
    begin = ...
    result = function()
    duration = ...
    logging.debug(....)
    return result

But it seems hard to refactor the inner code into nice functions that can be passed into log time.
Where are #DEFINES when you need them? :)